R/utilities-configuration-plan.R

```diff
@@ -117,12 +117,69 @@ getCurvePropertiesForTimeProfiles <- function(configurationPlanCurve,
 
   # Observed Data
   if (pathArray[2] %in% "ObservedData") {
-    observedData <- getObservedDataFromConfigurationPlan(pathArray[1], configurationPlan, logFolder)
+    observedResults <- getObservedDataFromConfigurationPlan(pathArray[1], configurationPlan, logFolder)
+    molWeight <- getMolWeightForObservedData(pathArray, configurationPlan, simulation)
+
+    time <- ospsuite::toUnit(
+      quantityOrDimension = "Time",
+      values = as.numeric(observedResults$data[, 1]),
+      targetUnit = axesProperties$x$unit,
+      sourceUnit = observedResults$metaData$time$unit
+    )
+    # Convert output values, if molWeight is NA but not required, then toUnit works without any issue
+    # if molWeight is NA and required, then toUnit crashes, error is caught
+    # and the error message indictes which observed data Id need molWeight
+    outputValues <- tryCatch({
+      ospsuite::toUnit(
+        quantityOrDimension = ospsuite::getDimensionForUnit(observedResults$metaData$output$unit),
+        values = observedResults$data[, 2],
+        targetUnit = axesProperties$y$unit,
+        sourceUnit = observedResults$metaData$output$unit,
+        molWeight = molWeight
+      )
+    },
+    error = function(e) {
+      NULL
+    }

```

Comment:
As written in the commented line, toUnit crashes **only if MW is NA and required**.
I used `tryCatch` to prevent the long unnecessary rClr message about toUnit error replaced by the following error message : "Molecular weight not found but required for observed data Id '<obs data id>' in Time Profile plot."